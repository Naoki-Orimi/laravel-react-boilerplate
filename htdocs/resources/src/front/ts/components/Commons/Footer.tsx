import * as React from "react";
import { connect } from "react-redux";

export class CommonFooter extends React.Component {
    constructor(props) {
        super(props);
    }

    render(): JSX.Element {
        return (
            <>
                <footer className="footer">
                    <div className="footer_inner">
                        <p className="mt20">
                            <img src="/assets/front/image/logo_02.png" alt="" />
                        </p>
                        <p className="mt10">
                            <small className="fz-s">
                                ©️isystk
                                このページは架空のページです。実際の人物・団体とは関係ありません。
                            </small>
                        </p>
                    </div>
                </footer>
            </>
        );
    }
}

const mapDispatchToProps = {};

export default connect(null, mapDispatchToProps)(CommonFooter);
