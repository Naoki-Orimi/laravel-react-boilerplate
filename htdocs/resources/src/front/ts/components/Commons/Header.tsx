import * as React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { URL } from "../../common/constants/url";

class CommonHeader extends React.Component {

  constructor(props) {
    super(props);
  }

  render(): JSX.Element {

    return (
      <React.Fragment>
      <header className="header shadow-sm">
          <nav className="navbar navbar-expand-md navbar-light bg-white headerNav">
              <a className="header_logo" href="/">
                  <img src="/assets/front/image/logo.png" alt="" className="" />
              </a>

              <div className="" id="navbarSupportedContent">
                  <ul className="navbar-nav mr-auto">

                  </ul>

                  <ul className="navbar-nav ml-auto">
                      <li className="nav-item">
                          <a className="btn btn-danger mr-3" href="/login">ログイン</a>
                      </li>
                      <li className="nav-item">
                          <a className="btn btn-link text-danger" href="/register">新規登録</a>
                      </li>
                      <li className="nav-item">
                          <a className="btn btn-link text-danger" href="#">
                              お問い合わせ
                          </a>
                      </li>

                      {/*
                      <li className="nav-item dropdown">
                          <a id="navbarDropdown" className="nav-link dropdown-toggle" href="#" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" v-pre>
                              {{ Auth::user()->name }} <span className="caret"></span>
                          </a>

                          <div className="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
                              <a className="dropdown-item" href="#" onclick="event.preventDefault();
                                                      document.getElementById('logout-form').submit();">
                                  {{ __('ログアウト') }}
                              </a>

                              <form id="logout-form" action="{{ route('logout') }}" method="POST" style="display: none;">
                                  @csrf
                              </form>

                              <a className="dropdown-item" href="#" onclick="event.preventDefault();
                                                      document.getElementById('shop-form').submit();">
                                  カートを見る
                              </a>

                              <form id="shop-form" action="{{ route('shop.mycart') }}" method="POST" style="display: none;">
                                  @csrf
                              </form>

                          </div>
                      </li>

                      <a href="#" onclick="event.preventDefault();
                                                      document.getElementById('shop-form').submit();">
                          <img src="{{ asset('/assets/front/image/cart.png') }}" className="cartImg ml-3">
                      </a>
                      <li className="nav-item">
                          <a className="btn btn-link text-danger" href="{{ route('contact.index') }}">
                              お問い合わせ
                          </a>
                      </li>

                       */}

                  </ul>
              </div>
          </nav>
      </header>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    parts: state.parts,
    auth: state.auth
  };
};

const mapDispatchToProps = {  };

export default connect(mapStateToProps, mapDispatchToProps)(CommonHeader);
