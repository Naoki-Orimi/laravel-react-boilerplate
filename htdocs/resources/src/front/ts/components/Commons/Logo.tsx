import React, { FC } from "react";

export const Logo: FC = () => {
    return (
        <a className="header_logo" href="/">
            <img src="/assets/front/image/logo.png" alt="" />
        </a>
    );
};

export default Logo;
