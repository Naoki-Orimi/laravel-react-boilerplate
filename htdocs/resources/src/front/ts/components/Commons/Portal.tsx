import ReactDOM from 'react-dom'
import React, { FC, useRef, useState, useEffect } from 'react'

const Portal: FC = ({ children }) => {
  const ref = useRef<HTMLElement>()
  const [mounted, setMounted] = useState(false)

  useEffect(() => {
    const current = document.querySelector<HTMLElement>('#react-root')
    if (current) {
      console.log('current', current)
      ref.current = current
    }
    setMounted(true)
  }, [])

  console.log('mounted', mounted)
  return mounted ? ReactDOM.createPortal(<>{children}</>, ref.current ? ref.current : new Element()) : null
}

export default Portal
