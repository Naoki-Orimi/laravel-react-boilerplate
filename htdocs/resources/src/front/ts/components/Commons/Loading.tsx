import React, { FC } from "react";
import Portal from "./Portal";
import MainService from "@/services/main";
import { Spinner } from "reactstrap";

type Props = {
    appRoot: MainService;
};

const Loading: FC<Props> = ({ appRoot }) => {
    const { isShowLoading } = appRoot;
    return (
        <Portal>
            {isShowLoading && (
                <div id="site_loader_overlay">
                    <div className="site_loader_spinner">
                        <Spinner>Loading...</Spinner>
                    </div>
                </div>
            )}
        </Portal>
    );
};

export default Loading;
