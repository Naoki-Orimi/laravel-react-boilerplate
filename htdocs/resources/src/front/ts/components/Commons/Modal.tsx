import React, { FC } from "react";
import { useDispatch, useSelector } from "react-redux";
import Portal from "./Portal";
import { Parts } from "@/stores/StoreTypes";
import { hideOverlay } from "@/services/actions";

type IRoot = {
    parts: Parts;
};

type Props = {
    children: React.ReactNode;
};

const Modal: FC<Props> = ({ children }) => {
    const dispatch = useDispatch();
    const { isShowOverlay } = useSelector<IRoot, Parts>(
        (state): Parts => state.parts
    );

    const onClose = e => {
        e.preventDefault();
        dispatch(hideOverlay());
    };

    return (
        <Portal>
            {isShowOverlay && <div id="overlay-background"></div>}
            <div className={`isystk-overlay ${isShowOverlay ? "open" : ""}`}>
                <button
                    type="button"
                    className="close"
                    aria-label="Close"
                    onClick={onClose}
                >
                    <span aria-hidden="true">&times;</span>
                </button>
                {children}
            </div>
        </Portal>
    );
};

export default Modal;
