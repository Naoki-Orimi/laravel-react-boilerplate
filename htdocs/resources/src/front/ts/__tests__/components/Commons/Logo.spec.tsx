import React from 'react';
import renderer from 'react-test-renderer';
import { Logo } from '../../../components/Commons/Logo';


test('Logo', () => {
  const component = renderer.create(
      <Logo />
  )
  expect(component).toMatchSnapshot()
})
