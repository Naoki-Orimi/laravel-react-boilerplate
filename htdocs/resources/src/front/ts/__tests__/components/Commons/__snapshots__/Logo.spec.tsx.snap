// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo 1`] = `
<a
  className="header_logo"
  href="/"
>
  <img
    alt=""
    src="/assets/front/image/logo.png"
  />
</a>
`;
